src/caip-chain-id.ts

```diff
@@ -0,0 +1,65 @@
+import { is, pattern, string, Struct } from 'superstruct';
+
+import { assert } from './assert';
+
+export type CaipChainId = `${string}:${string}`;
```

Comment:
Typescript doesn't give us much value here

Reply:
Yeah, unfortunately, not much further we can go here, but I think just having a type for this is useful from a naming perspective, so it's okay to me.